# Double: multiplier to apply to the maximum allowable vehicle deceleration limit so we plan under our capabilities
vehicle_decel_limit_multiplier : 1.0

# Double: multiplier to apply to the maximum allowable vehicle acceleration limit so we plan under our capabilities
vehicle_accel_limit_multiplier : 1.0

# Double: The minimum distance in meters that the vehicle can be at before requiring a transition to the APPROACH state
min_approach_distance : 30.0

# Double: Downtrack distance until nearest intersection where the Trajectory Smoothing algorithm should activate
trajectory_smoothing_activation_distance: 140.0

# Double: A buffer infront of the stopping location which will still be considered a valid stop. Units in meters
stopping_location_buffer : 4.0

# Double: A buffer in seconds around the green phase which will reduce the phase length such that
#         vehicle still considers it non-green. For example, having 1s buffer means vehicle will
#         only consider it green 1s after the physical signal turned green and before its physical
#         green ends. This parameter exists to capture the uncertainty where vehicle may not be able
#         to perfectly follow the planned trajectory and targeting a buffered entry time helps it
#         enter at the right time physically.
# NOTE: The value used internally is actually the max of this value and `vehicle_response_lag` set
#       in /opt/carma/vehicle/calibration/identifiers/UniqueVehicleParams.yaml. Since
#       green_light_time_buffer is meant to accommodate the uncertainty of vehicle response to the
#       planned trajectory, it should be at least or more the value of `vehicle_response_lag`, which
#       is also a controller parameter that intends to compensate for the uncertainty in the
#       physical vehicle accurately following a trajectory
green_light_time_buffer : 0.5

# Double: Minimum allowable speed in trajectory smoothing algorithm m/s
algo_minimum_speed : 4.4704

# Double: Safety multiplier (must be less than 1.0) of planned allowable vehicle deceleration to use when stopping. This new deceleration makes vehicle decelerate earlier distance.
#         NOTE: Stacks on vehicle_decel_limit_multiplier and stopping uses max_decel; this distance is only used for calculating earlier downtrack
deceleration_fraction : 0.7

# Double: Desired distance to stop buffer in meters
desired_distance_to_stop_buffer : 1.0

# Double: The minimum period in seconds which a maneuver plan must cover if the plugin wishes to control the whole plan
min_maneuver_planning_period : 15.1

# Bool: If enable_carma_streets_connection is true when we want to allow carma streets functionality (UC3) and if its false that means we don't want to allow carma streets behaviour and will only use UC2 behaviour.
enable_carma_streets_connection : false

# Int: 0 to turn off and 1 turn on SPAT processor
enable_carma_wm_spat_processing: 1

# Double: Mobility operation rate
mobility_rate : 10.0

# String: The name to use for this plugin during comminications with the arbitrator
strategic_plugin_name : lci_strategic_plugin

# String: The name of the tactical plugin to use for Lane Following trajectory planning
#         This plugin is used to apply trajectory smoothing algorithm BEFORE entering the intersection if within activation distance
lane_following_plugin_name : light_controlled_intersection_tactical_plugin

# String: The name of the plugin to use for stop and wait trajectory planning
stop_and_wait_plugin_name : stop_and_wait_plugin

# String: The name of the plugin to use for intersection transit trajectory planning
#         This plugin is used to travel INSIDE the intersection where there is no trajectory smoothing algorithm active
intersection_transit_plugin_name : intersection_transit_maneuvering
